contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -27,31 +27,88 @@ RETURNS OID
 AS 'babelfishpg_tsql', 'user_id'
 LANGUAGE C IMMUTABLE PARALLEL SAFE;
 
-CREATE OR REPLACE FUNCTION sys.suser_name(IN server_user_id OID DEFAULT NULL)
+CREATE OR REPLACE FUNCTION sys.suser_name_internal(IN server_user_id OID)
 RETURNS sys.NVARCHAR(128)
 AS 'babelfishpg_tsql', 'suser_name'
 LANGUAGE C IMMUTABLE PARALLEL RESTRICTED;
 
+CREATE OR REPLACE FUNCTION sys.suser_name(IN server_user_id OID)
+RETURNS sys.NVARCHAR(128) AS $$
+    SELECT CASE 
+        WHEN server_user_id IS NULL THEN NULL
+        ELSE sys.suser_name_internal(server_user_id)
+    END;
+$$
+LANGUAGE SQL IMMUTABLE PARALLEL RESTRICTED;
+
+CREATE OR REPLACE FUNCTION sys.suser_name()
+RETURNS sys.NVARCHAR(128)
+AS $$
+    SELECT sys.suser_name_internal(NULL);
+$$
+LANGUAGE SQL IMMUTABLE PARALLEL RESTRICTED;
+
 -- Since SIDs are currently not supported in Babelfish, this essentially behaves the same as suser_name but 
 -- with a different input data type
-CREATE OR REPLACE FUNCTION sys.suser_sname(IN server_user_sid SYS.VARBINARY(85) DEFAULT NULL)
+CREATE OR REPLACE FUNCTION sys.suser_sname(IN server_user_sid SYS.VARBINARY(85))
 RETURNS SYS.NVARCHAR(128)
 AS $$
     SELECT sys.suser_name(CAST(server_user_sid AS INT)); 
 $$
 LANGUAGE SQL IMMUTABLE PARALLEL RESTRICTED;
 
-CREATE OR REPLACE FUNCTION sys.suser_id(IN login TEXT DEFAULT NULL)
+CREATE OR REPLACE FUNCTION sys.suser_sname()
+RETURNS SYS.NVARCHAR(128)
+AS $$
+    SELECT sys.suser_name();
+$$
+LANGUAGE SQL IMMUTABLE PARALLEL RESTRICTED;
+
+CREATE OR REPLACE FUNCTION sys.suser_sname()
```

Comment:
Function definition is redundant for suser_sname(). Please remove one out it.

Reply:
Whoops, my bad. Removed in the newest commit!

Reply:
Looks like you haven't pushed that change yet. Could you please check once?

Reply:
Hi @shalinilohia50 , I think for this one I deleted the duplicate underneath (so it may not be showing the file as outdated).  The changes should be shown though under file changes.

Reply:
Ack, my bad on that one. The duplicate should have been removed now.